lib/src/screens/courses/specificSectionView.dart

```diff
@@ -1,89 +1,109 @@
-    semester: Semester.Fall,
-    semesterID: "fa",
-    subject: 'Computer Science',
-    subjectID: "CS",
-    courseID: 0,
-    title: 'Introduction to Computer Science I',
-    description:
-        'Basic concepts in computing and fundamental techniques for solving computational problems. Intended as a first course for computer science majors and others with a deep interest in computing. Credit is not given for both CS 124 and CS 125. Prerequisite: Three years of high school mathematics or MATH 112.',
-    creditHours: '3 hours.',
-    courseSectionInformation:
-        'Credit is not given for both CS 124 and CS 125. Prerequisite: Three years of high school mathematics or MATH 112.',
-    classScheduleInformation: null,
-    sections: [],
-    categories: [],
-  );
+  List<HomeworkItem>? homework;
+  HomeworkAPI homeworkApi = HomeworkAPI();
+  CourseAPI courseApi = CourseAPI();
+  late Future<SectionItem?> section;
+  late Future<CourseItem?> course;
 
-  List<Widget> _getDetails(SectionItem? section) {
-    List<HomeworkItem> homework = [
-      HomeworkItem(
-        dueDate: DateTime.now().add(Duration(days: 2)),
-        name: 'Practice Problems #1',
-        description: 'This homework will help prepare you for the test!',
-        assignmentUrl: 'https://example.com',
-        platform: 'turnitin',
-        course: course,
-        files: [
-          File(
-            name: 'Problem set.json',
-            mimeType: 'application/json',
-            size: 400,
-            url: 'https://example.com',
-          ),
-          File(
-            name: 'Problem set.json',
-            mimeType: 'application/json',
-            size: 400,
-            url: 'https://example.com',
-          ),
-          File(
-            name: 'Problem set.json',
-            mimeType: 'application/json',
-            size: 400,
-            url: 'https://example.com',
-          )
-        ],
-      ),
-      HomeworkItem(
-        name: 'Practice Problems #2',
-        dueDate: DateTime.now().add(Duration(days: 7)),
-        assignmentUrl: 'https://en.wikipedia.org/wiki/Hot_air_ballooning',
-        platform: 'turnitin',
-        course: course,
-      )
-    ];
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    var api = ApiServiceWidget.of(context).api;
 
+    if (widget.sectionCode != null) {
+      var courseKeyword = widget.sectionCode!.split('_')[0];
+      courseKeyword =
+          '${courseKeyword.substring(0, 2)} ${courseKeyword.substring(2)}';
+      setState(() {
+        section = courseApi.getSection(api, widget.sectionCode!);
+        course = courseApi
+            .getCourses(api, courseKeyword)
+            .then((courses) => courses![0]);
+      });
+    } else {
+      setState(() {
+        section = Future.value(widget.section);
+        course = Future.value(widget.course);
+      });
+    }
+
+    course.then((course) async {
+      var homework = await homeworkApi.getHomework('code');
+
```

Comment:
```suggestion
if (!mounted) return;
```
Make sure the widget is still mounted when setting state after this await.